code/website/site.hs

```diff
@@ -11,30 +11,52 @@ import System.Directory (doesDirectoryExist, doesFileExist, listDirectory)
 import System.Environment (getEnv, lookupEnv)
 import System.FilePath (takeBaseName, takeExtension)
 
+-- type FilePath = String -- from System.FilePath
 type Name = String
 type SourceLocation = Maybe String
 type SRSVariants = [(Name, String)]
-data Example = E Name SourceLocation SRSVariants
+type Description = Maybe String
+data Example = E Name SourceLocation SRSVariants Description
 
+-- Returns FilePath of the SRS
 srsPath :: FilePath -> FilePath -> FilePath -> FilePath
 srsPath baseDir ex srs = baseDir ++ ex ++ "/" ++ srs
 
+-- Returns the Uppercase String of the given extension
 getExtensionName :: String -> String
 getExtensionName [] = error "Expected some file extension. Got none."
 getExtensionName [_] = error "Expected file extension. Got a single character."
 getExtensionName ('.':xs) = map toUpper xs
 getExtensionName _ = error "Expected some extension."
 
+-- returns a list of tuples, where each tuple is (Name, string) where string is "HTML" or "PDF"
 getSRS :: [Name] -> SRSVariants
 getSRS = map (\x -> (x, getExtensionName $ takeExtension x)) . filter (\x -> any (`endswith` x) [".pdf", ".html"])
 
+zip4 :: [a] -> [b] -> [c] -> [d] -> [(a,b,c,d)]
+zip4 (a:as) (b:bs) (c:cs) (d:ds) = (a,b,c,d) : zip4 as bs cs ds
+zip4 _ _ _ _ = []
+
```

Comment:
[Builtin to base](https://hackage.haskell.org/package/base-4.7.0.0/docs/Data-List.html#v:zip4)

Reply:
Will remove and use built in version.